pkg/backup/item_collector.go

```diff
@@ -275,58 +274,6 @@ func (r *itemCollector) getResourceItems(log logrus.FieldLogger, gv schema.Group
 
 	namespacesToList := getNamespacesToList(r.backupRequest.NamespaceIncludesExcludes)
 
-	// Check if we're backing up namespaces for a less-than-full backup.
```

Comment:
I agree with @ywk253100 to remove this part of code, because the most of namespace handling logic is duplicate with the following other resources handling logic.
And the current namespace handling logic cannot work with OrLabelSelector.

Reply:
@blackpiglet I think this should be fine. So if I'm understanding the code correctly, this means if a backup includes "ns1" and "ns2" and adds a label selector that only includes items in "ns1", then we back up both namespaces anyway. With current velero behavior, this won't matter for restore since empty namespaces aren't restored (as described in https://github.com/vmware-tanzu/velero/issues/1970 ) -- if the resolution of that issue is to restore empty namespaces, then the effect here would be that "ns1" would be restored with labeled resources inside, an "ns2" would be restored empty.